src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -384,9 +384,13 @@ protected override Symbol MakePatternVariable(DeclarationPatternSyntax node, Syn
         {
             throw ExceptionUtilities.Unreachable;
         }
+
         protected override Symbol MakeOutVariable(DeclarationExpressionSyntax node, BaseArgumentListSyntax argumentListSyntax, SyntaxNode nodeToBind)
         {
-            return SourceMemberFieldSymbolFromDesignation.Create(_containingType, node.VariableDesignation(), _modifiers, _containingFieldOpt, nodeToBind);
+            var single = node.VariableDesignation();
+
+            return SourceMemberFieldSymbolFromDesignation.Create(_containingType, single, ((TypedVariableComponentSyntax)single.Parent).Type,
```

Comment:
>((TypedVariableComponentSyntax)single.Parent).Type [](start = 90, length = 50)

There is an extension method named Type to get type syntax out of DeclarationExpressionSyntax